NextJS @nx/devkit build failure

### Current Behavior

As of v17.1.2, it is not possible to build nextjs projects, as the generator file for @nx/next is looking for a non-existent type of
`import type { ProjectNameAndRootFormat } from '@nx/devkit/src/generators/project-name-directory-utils';
`
Attempting to build with this version will result in the following error:
`Type error: Cannot find module '@nx/devkit/src/generators/project-name-directory-utils' or its corresponding type declarations.`

This is for the file: @nx/next/src/generators/library/schema.d.ts

I believe this should be resolved by changing this type import to the following:
`import type { ProjectNameAndRootFormat } from '@nx/devkit/src/generators/project-name-and-root-utils';
`

### Expected Behavior

running nx build [app-name] with an application that uses NextJS successfully compiles the production nextJS build into the specified directory, as per the nx project configuration.

### GitHub Repo

_No response_

### Steps to Reproduce

1. Navigate to NX root
2. Run nx build [application-name]
3. Observe the terminal output


### Nx Report

```shell
>  NX   Report complete - copy this into the issue template

   Node   : 20.9.0
   OS     : darwin-arm64
   npm    : 10.1.0
   
   nx (global)        : 17.1.2
   nx                 : 17.1.2
   @nx/js             : 17.1.2
   @nx/jest           : 17.1.2
   @nx/linter         : 17.1.2
   @nx/eslint         : 17.1.2
   @nx/workspace      : 17.1.2
   @nx/cypress        : 17.1.2
   @nx/devkit         : 17.1.2
   @nx/eslint-plugin  : 17.1.2
   @nx/next           : 17.1.2
   @nx/react          : 17.1.2
   @nrwl/tao          : 17.1.2
   @nx/web            : 17.1.2
   typescript         : 5.2.2
```


### Failure Logs

```shell
> nx run [redacted]-admin:build:production --no-cache

- info Loaded env from /Users/[redacted]/Projects/[redacted]/Working/[redacted]/modules/[redacted]-frontend-nx/apps/[redacted]-admin/.env.local
- info Creating an optimized production build...
- info Compiled successfully
- info Skipping linting
- info Checking validity of types...
- warn TypeScript project references are not fully supported. Attempting to build in incremental mode.
Failed to compile.

../../node_modules/@nx/next/src/generators/library/schema.d.ts:1:47
Type error: Cannot find module '@nx/devkit/src/generators/project-name-directory-utils' or its corresponding type declarations.
> 1 | import type { ProjectNameAndRootFormat } from '@nx/devkit/src/generators/project-name-directory-utils';
    |                                               ^
  2 | import type { Linter } from '@nx/eslint';
  3 | import type { SupportedStyles } from '@nx/react';
  4 | 
Error occurred while trying to run the build command
1

 ————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————

 >  NX   Ran target build for project [redacted]-admin (18s)
 
         With additional flags:
           --cache=false
 
    ✖    1/1 failed
    ✔    0/1 succeeded [0 read from cache]
```


### Package Manager Version

10.1.0

### Operating System

- [X] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information

I have resolved this successfully by manually changing the import from
`import type { ProjectNameAndRootFormat } from '@nx/devkit/src/generators/project-name-directory-utils';
`
to
`import type { ProjectNameAndRootFormat } from '@nx/devkit/src/generators/project-name-and-root-utils';
`

But was unable to open a PR / branch with the hotfix due to permissions.
I hope you resolve this for NextJS users soon.

--- a/packages/next/src/generators/library/schema.d.ts
+++ b/packages/next/src/generators/library/schema.d.ts
@@ -1,4 +1,4 @@
-import type { ProjectNameAndRootFormat } from '@nx/devkit/src/generators/project-name-directory-utils';
+import type { ProjectNameAndRootFormat } from '@nx/devkit/src/generators/project-name-and-root-utils';
 import type { Linter } from '@nx/eslint';
 import type { SupportedStyles } from '@nx/react';
 

--- a/packages/plugin/src/generators/create-package/schema.d.ts
+++ b/packages/plugin/src/generators/create-package/schema.d.ts
@@ -1,4 +1,4 @@
-import type { ProjectNameAndRootFormat } from '@nx/devkit/src/generators/project-name-directory-utils';
+import type { ProjectNameAndRootFormat } from '@nx/devkit/src/generators/project-name-and-root-utils';
 import type { Linter } from '@nx/eslint';
 
 export interface CreatePackageSchema {

--- a/packages/plugin/src/generators/e2e-project/schema.d.ts
+++ b/packages/plugin/src/generators/e2e-project/schema.d.ts
@@ -1,4 +1,4 @@
-import type { ProjectNameAndRootFormat } from '@nx/devkit/src/generators/project-name-directory-utils';
+import type { ProjectNameAndRootFormat } from '@nx/devkit/src/generators/project-name-and-root-utils';
 import type { Linter } from '@nx/eslint';
 
 export interface Schema {

--- a/packages/plugin/src/generators/plugin/schema.d.ts
+++ b/packages/plugin/src/generators/plugin/schema.d.ts
@@ -1,4 +1,4 @@
-import type { ProjectNameAndRootFormat } from '@nx/devkit/src/generators/project-name-directory-utils';
+import type { ProjectNameAndRootFormat } from '@nx/devkit/src/generators/project-name-and-root-utils';
 import type { Linter } from '@nx/eslint';
 
 export interface Schema {

